pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -40,14 +40,36 @@ const (
 
 // Returns a configProcessingSkipFlag
 func parseConfig(config *zconfig.EdgeDevConfig, getconfigCtx *getconfigContext,
-	usingSaved bool) bool {
+	source configSource) bool {
 
 	// Do not accept new commands from Local profile server while new config
 	// from the controller is being applied.
-	getconfigCtx.localCommands.Lock()
-	defer getconfigCtx.localCommands.Unlock()
+	if getconfigCtx.localCommands != nil {
+		getconfigCtx.localCommands.Lock()
+		defer getconfigCtx.localCommands.Unlock()
+	}
+
+	// Update lastReceivedConfig even if the config processing is skipped.
+	if config.ConfigTimestamp.IsValid() {
+		getconfigCtx.lastReceivedConfig = config.ConfigTimestamp.AsTime()
+	} else {
+		getconfigCtx.lastReceivedConfig = time.Now()
```

Comment:
Odd to set lastReceivedConfig in the case where it is ignored due to being too old, since it would typically be overwritten (by the next getconfig). Move this block after the obsolete check?

Reply:
true, moved the blocked after the skip